CRM/Core/Payment.php

```diff
@@ -1286,9 +1286,24 @@ public function doPayment(&$params, $component = 'contribute') {
    *
    * @param array $params
    *
+   * @return array
    * @throws \Civi\Payment\Exception\PaymentProcessorException
    */
-  public function doRefund(&$params) {}
+  public function doRefund($params) {
```

Comment:
I have created an issue for discussion / approval of this change https://lab.civicrm.org/dev/financial/issues/92